src/controllers/controller.bar.js

```diff
@@ -397,17 +407,23 @@ module.exports = function(Chart) {
 						imeta.stack === stack &&
 						imeta.controller.getValueScaleId() === scale.id &&
 						chart.isDatasetVisible(i)) {
+						yStackValue = scale.parseValue(datasets[i].data[index]);
 
-						ivalue = scale.getRightValue(datasets[i].data[index]);
-						if ((value < 0 && ivalue < 0) || (value >= 0 && ivalue > 0)) {
+						if (yStackValue.min === yStackValue.max) {
+							ivalue = yStackValue.val;
+						} else {
+							ivalue = yStackValue.val <= 0 ? yStackValue.min : yStackValue.max;
+						}
+
+						if ((value.min < 0 && ivalue < 0) || (value.max >= 0 && ivalue > 0)) {
 							start += ivalue;
 						}
 					}
 				}
 			}
 
 			base = scale.getPixelForValue(start);
-			head = scale.getPixelForValue(start + value);
+			head = scale.getPixelForValue(start + value.val);
```

Comment:
shouldn't this be `max` instead of `val`?

Reply:
no-no-no, for D implementation, it should start value + plus value.val which is defference between max and min values. Check this - http://pravopys.net/chartjs/samples/charts/bar/stacked-vertical-simon.html

Reply:
in that case you could replace `start + value.val` with `value.max`. because right now `start` will always be min which means this equals min + max - min